src/cms-driver.ts

```diff
@@ -18,6 +18,20 @@ import * as util from './util';
 
 const log = createLogger('cms-driver');
 
+class CMakeInputFile {
```

Comment:
Please move the cmake dirty check into a separate file, it is an encapsulated unit.